### DISCUSSION TITLE
Good practices voor examen programmeren

### DISCUSSION THREAD

**Discussion by IngeVerbeek** [dodona-edu/feedback-and-support#49] [Questions - Vragen]
Binnenkort hebben onze leerlingen (6e middelbaar) voor het eerst een examen informaticawetenschappen, inclusief een aantal programmeeropgaven in python. We hebben tijdens het schooljaar al met Dodona gewerkt en ook de toetsen zo georganiseerd.

Wij zijn ongetwijfeld niet de eerste die een examen programmeren met Dodona organiseren. Misschien heeft er iemand tips en tricks, dingen die wel/niet goed lopen, ... We hebben sinds kort "meeleessoftware" en kunnen daarmee de toegang tot bepaalde applicaties en/of websites blokkeren, dus dat zullen we zeker gebruiken. 

Bijvoorbeeld: 
- een aparte cursus aanmaken voor het examen of in dezelfde cursus verder werken (maar dan hebben ze ook toegang tot hun vorige opgaven?)
- laten jullie de leerlingen in een IDE werken (bij ons: Thonny) of moeten ze rechtstreeks in Dodona programmeren?
- ...

**Comment by marcothijs** [comment]
Dag Inge, 

Ik doe mijn Python toetsen en examens (5e en 6e jaar Applicatie- en databeheer en voorheen Industriële ICT) al 4 jaar in Dodona maar ik begrijp je bezorgdheden. Eerst en vooral zijn al mijn praktische evaluaties openboek, dus ze mogen de cursus, hun notities en de oefeningen uit Dodona gebruiken. Ik maak meestal een nieuwe reeks aan per toets of examen binnen de cursus zelf die ik openzet tijdens en daarna ook weer sluit. Na de evaluatie (gedaan in Dodona) zet ik ze terug open met de feedback op hun oplossingen. Qua IDE's laat ik enkel de standaard Python IDLE of Dodona zelf toe. Andere IDE zoals Pycharm en VSC hebben AI-add-ons en/of een collaboratie-functie en laat ik dus niet meer toe. Waar het vóór het ChatGPT-tijdperk de elektronisch communicatie tussen leerlingen via Discord een probleem vormde zijn het nu de web AI-tools zoals ChatGPT die het voor de leerling zeer gemakkelijk maakt om snel een oplossing te vinden en die te plakken in Dodona. Sinds het bestaan van ChatGPT heb ik al meerdere leerlingen betrapt op het gebruik ervan maar zonder meeleessoftware blijft het moeilijk dit hard te maken (ik laat ChatGPT de opgave op voorhand al eens oplossen). Onze school is nu een pilotproject gestart met classroom cloud om onder andere dit probleem aan te pakken. De plagiaat detecteren functie in Dodona gebruik ik eerder sporadisch omdat bij korte en eenvoudige oefeningen dit weinig zegt.

M.v.g.
Marco
